### PR TITLE
[dotnet] Add migration about serialization

### PR DESCRIPTION
**Pull request by bruno-garcia**
Relates to: https://github.com/getsentry/sentry-dotnet/issues/962

### CODE REVIEW

File: src/platforms/dotnet/migration.mdx [hunk 1]
```diff
@@ -22,6 +22,51 @@ One big caveat is that this will **affect grouping!**. If you prefer to stay wit
 options.StackTraceMode = StackTraceMode.Original;
 ```
 
+### Serialization
+
+On version 3.0.0 the Sentry SDK for .NET changed from `Newtonsoft.Json` to `System.Text.Json`.
+This introduces some advantages, such as no dependency is needed (when targeting .NET Core 3.0, .NET 5 and later version) and much faster performance.
+
+One breaking change in behavior that came up was that `Newtonsoft.Json` was able to serialize some types that are not possible with `System.Text.Json`.
+This affects objects set as [Context](../enriching-events/context/) and `Extra`.
```
Comment (imatwawana) [inline src/platforms/dotnet/migration.mdx:31]:
```suggestion
One breaking change that affects objects set as [Context](../enriching-events/context/) and `Extra` is that `Newtonsoft.Json` was previously able to serialize some types, which is not possible with `System.Text.Json`.
```

**Review by imatwawana — APPROVED** [review]
Minor wording edits for clarity/style. One area where there's a question. Otherwise LGTM.

File: src/platforms/dotnet/migration.mdx [hunk 2]
```diff
@@ -22,6 +22,51 @@ One big caveat is that this will **affect grouping!**. If you prefer to stay wit
 options.StackTraceMode = StackTraceMode.Original;
 ```
 
+### Serialization
+
+On version 3.0.0 the Sentry SDK for .NET changed from `Newtonsoft.Json` to `System.Text.Json`.
+This introduces some advantages, such as no dependency is needed (when targeting .NET Core 3.0, .NET 5 and later version) and much faster performance.
```
[the remark this answers is not in the document]
Comment (bruno-garcia) [reply src/platforms/dotnet/migration.mdx:28]: I was hoping to describe that the SDK has "no dependency". Let me see if I can reword it:

Comment (bruno-garcia) [inline src/platforms/dotnet/migration.mdx:28]:
```suggestion
Using the new library has many advantages. One of which is the lack of an additional dependency (when targeting .NET Core 3, .NET 5 and later versions) and much faster performance.
```

[the remark this answers is not in the document]
Comment (bruno-garcia) [reply src/platforms/dotnet/migration.mdx]: Is the new version a bit better?

File: src/platforms/dotnet/migration.mdx [hunk 3]
```diff
@@ -24,13 +24,12 @@ options.StackTraceMode = StackTraceMode.Original;
 
 ### Serialization
 
-On version 3.0.0 the Sentry SDK for .NET changed from `Newtonsoft.Json` to `System.Text.Json`.
-This introduces some advantages, such as no dependency is needed (when targeting .NET Core 3.0, .NET 5 and later version) and much faster performance.
+In version 3.0.0, the Sentry SDK for .NET changed from `Newtonsoft.Json` to `System.Text.Json`.
+Using the new library has many advantages. One of which is the lack of an additional dependency (when targeting .NET Core 3, .NET 5 and later versions) and much faster performance.
```
Comment (imatwawana) [inline src/platforms/dotnet/migration.mdx:28]:
```suggestion
Using the new library has many advantages. These include the lack of an additional dependency (when targeting .NET Core 3, .NET 5 and later versions) and much faster performance.
```

**Pull request merged**